src/main/java/com/dev/cinema/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.dev.cinema.service.impl;
+
+import com.dev.cinema.dao.OrderDao;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.Order;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.OrderService;
+import com.dev.cinema.service.ShoppingCartService;
+import java.util.List;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(List<Ticket> tickets, User user) {
+        Order order = new Order();
+        order.setTickets(tickets);
+        order.setUser(user);
+        shoppingCartService.clear(shoppingCartService.getByUser(user));
+        return orderDao.add(order);
```

Comment:
should we add orderDate when complete order?

Reply:
let's pass list of tickets and user to this method in order not to create dependency shopping cart - order